### ISSUE TITLE
Diagnostics file is not written when duplicate keys are detected

### ISSUE THREAD

**Issue by mikeminutillo** [Particular/NServiceBus#5669]
This is [by design](https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Hosting/StartupDiagnostics/HostStartupDiagnosticsWriter.cs#L20-L29) but it is not ideal.

When it happens, we are missing one of the primary tools that can be used to diagnose problems and usually there is not much that the customer can do about it.

Would it be better to turn the diagnostics into a list of key-value pairs instead of a dictionary before we serialize it? Or maybe add a prefix/suffix to de-duplicate keys?

## Example

There was a customer doing this:

```csharp
var outboxPersistence = endpointConfiguration.UsePersistence<SqlPersistence, StorageType.Outbox>();
outboxPersistence.SqlDialect<SqlDialect.MsSqlServer>().Schema("persistence");
// .. configuration on outboxPersistence

var sagaPersistence = endpointConfiguration.UsePersistence<SqlPersistence, StorageType.Sagas>();
sagaPersistence.SqlDialect<SqlDialect.MsSqlServer>().Schema("persistence");
// .. configuration on sagaPersistence

var timeoutPersistence = endpointConfiguration.UsePersistence<NHibernatePersistence, StorageType.Timeouts>();
// .. configuration of timeoutPersistence
```

When they encountered an error was the first time they looked for a startup diagnostics file, but it was not present. Instead they had this in the logs:

> Error | NServiceBus.WriteStartupDiagnostics | Diagnostics entries contains duplicates. Duplicates: NServiceBus.Persistence.Sql.SqlDialect

**Comment by mikeminutillo** [comment]
NOTE: The problem that causes this specfically in SQL Persistence was fixed here https://github.com/Particular/NServiceBus.Persistence.Sql/issues/349 but I still think that it would be better to write the diagnostic file in these cases.

**Comment by mikeminutillo** [comment]
[Released in 7.4](https://github.com/Particular/NServiceBus/releases/tag/7.4.0)
